org.jacoco.cli/src/org/jacoco/cli/internal/commands/Report.java

```diff
@@ -0,0 +1,170 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2017 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    John Keeping - initial implementation
+ *    Marc R. Hoffmann - rework
+ *
+ *******************************************************************************/
+package org.jacoco.cli.internal.commands;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.jacoco.cli.internal.Command;
+import org.jacoco.core.analysis.Analyzer;
+import org.jacoco.core.analysis.CoverageBuilder;
+import org.jacoco.core.analysis.IBundleCoverage;
+import org.jacoco.core.analysis.IClassCoverage;
+import org.jacoco.core.data.ExecutionDataStore;
+import org.jacoco.core.tools.ExecFileLoader;
+import org.jacoco.report.DirectorySourceFileLocator;
+import org.jacoco.report.FileMultiReportOutput;
+import org.jacoco.report.IReportVisitor;
+import org.jacoco.report.ISourceFileLocator;
+import org.jacoco.report.MultiReportVisitor;
+import org.jacoco.report.MultiSourceFileLocator;
+import org.jacoco.report.csv.CSVFormatter;
+import org.jacoco.report.html.HTMLFormatter;
+import org.jacoco.report.xml.XMLFormatter;
+import org.kohsuke.args4j.Argument;
+import org.kohsuke.args4j.Option;
+
+/**
+ * The <code>report</code> command.
+ */
+public class Report extends Command {
+
+	@Argument(usage = "list of JaCoCo *.exec files to read", metaVar = "<execfiles>")
+	List<File> execfiles = new ArrayList<File>();
```

Comment:
@marchof I have a fear that an empty list will be a common source of mistakes that lead to a report that shows zeros. However at least in case of HTML report we can improve later the report itself for such situations, but unfortunately not other formats. So wondering if we should print at least a warning? However maybe will be better to ban this because of existence of `quiet` option? Especially that there is command `classinfo` that just runs analysis of class files without exec files.

BTW seems that in this case not possible to navigate lower than method level or switch view to sources even when they are provided, but this is probably something unrelated to this PR:
```
java -jar /tmp/jacococli.jar report -classfiles /tmp/jacococli.jar -sourcefiles src -html /tmp/report
```

Also maybe a separate question, but I'm wondering about correctness of a behavior when destination already exists?